relational_operators/DestroyHashOperator.cpp

```diff
@@ -32,14 +32,23 @@ bool DestroyHashOperator::getAllWorkOrders(
     tmb::MessageBus *bus) {
   if (blocking_dependencies_met_ && !work_generated_) {
     work_generated_ = true;
-    container->addNormalWorkOrder(new DestroyHashWorkOrder(hash_table_index_, query_context),
-                                  op_index_);
+    container->addNormalWorkOrder(
+        new DestroyHashWorkOrder(input_relation_, hash_table_group_index_, query_context, is_numa_aware_join_),
+        op_index_);
   }
   return work_generated_;
 }
 
 void DestroyHashWorkOrder::execute() {
-  query_context_->destroyJoinHashTable(hash_table_index_);
+  if (input_relation_.hasPartitionScheme() && is_numa_aware_join_) {
+    const std::size_t num_partitions =
+        input_relation_.getPartitionScheme().getPartitionSchemeHeader().getNumPartitions();
```

Comment:
If I understand correctly, what you need from `input_relation_` is `num_partitions`. So please add a field in `DestroyHashWorkOrder` proto, and set the value using `DestroyHashWorkOrder` constructor.